WinFormsUI/MainForm.cs

```diff
@@ -9,12 +9,9 @@ namespace WinFormsUI
     public partial class MainForm : Form
     {
         Date startDate, endDate;
-        event Action OnFilteringUpdated;
+        event Action OnFilteringUpdated = () => { };
```

Comment:
Why we need this?

Reply:
To ensure that OnFilteringUpdated is never null and don't check for that before invoking it.